apollo-router/tests/samples/basic/interface-object/plan.json

```diff
@@ -0,0 +1,219 @@
+                                }
+                            },
+                            "response": {
+                                "body": {
+                                    "data": {
+                                        "req": {
+                                            "__typename": "C",
+                                            "id": "1",
+                                            "i": {
+                                                "__typename": "A",
+                                                "id": "1",
+                                                "x": 1
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    ],
+                    "products": {
+                        "requests": [
+                            {
+                                "request": {
+                                    "body": {
+                                        "query": "query TestItf2__products__1($representations:[_Any!]!){_entities(representations:$representations){...on I{y}}}",
+                                        "operationName": "TestItf2__products__1",
+                                        "variables": {
+                                            "representations": [
+                                                {
+                                                    "__typename": "I",
+                                                    "id": "1"
+                                                }
+                                            ]
+                                        }
+                                    }
+                                },
+                                "response": {
+                                    "body": {
+                                        "data": {
+                                            "_entities": [
+                                                {
+                                                    "y": 1
+                                                }
+                                            ]
+                                        }
+                                    }
+                                }
+                            }
+                        ]
+                    },
+                    "reviews": {
+                        "requests": []
+                    }
+                }
+            }
+        },
+        {
+            "type": "Request",
+            "request": {
+                "query": "query TestItf2 { req { id c } }"
+            },
+            "expected_response": {
+                "data": {
+                    "req": {
+                        "id": "1",
+                        "c": "a"
+                    }
+                }
+            }
+        },
```

Comment:
this is an interesting failure. This is what it logs:

```
query: {"query":"query TestItf2 { req { id c } }"}

subgraphs received requests:
        method: POST
        path: http://localhost/
        {"query":"query TestItf2__accounts__0{req{__typename id i{__typename id x}}}","operationName":"TestItf2__accounts__0"}

        method: POST
        path: http://localhost/
        {"query":"query TestItf2__products__1($representations:[_Any!]!){_entities(representations:$representations){...on I{y}}}","operationName":"TestItf2__products__1","variables":{"representations":[{"__typename":"I","id":"1"}]}}

assertion `left == right` failed
expected: {"data":{"req":{"id":"1","c":"a"}}}
received: {"data":{"req":{"id":"1","c":null}},"errors":[{"message":"HTTP fetch failed from 'reviews': 400: Bad Request","path":["req"],"extensions":{"code":"SUBREQUEST_HTTP_ERROR","service":"reviews","reason":"400: Bad Request","http":{"status":400}}},{"message":"Unknown type \"C\".","path":["req"],"extensions":{"code":"GRAPHQL_VALIDATION_FAILED"}}]}
```

The `reviews` subgraph should have received the query `query TestItf2__reviews__2($representations:[_Any!]!){_entities(representations:$representations){...on C{c}}}` with variables `{"representations": [{"__typename": "C"), "i": {"x": 1, "y": null}, "id": "1"}]}`, so this means that something is validating graphql queries between the moment the subgraph request is created, and the moment it reaches the subgraph

Reply:
this is the subgraph schema that was extracted by the planner for `reviews`:

```graphql
schema
  @link(url: "https://specs.apollo.dev/link/v1.0")
  @link(url: "https://specs.apollo.dev/federation/v2.9", import: ["@key", "@requires", "@provides", "@external", "@tag", "@extends", "@shareable", "@inaccessible", "@override", "@composeDirective", "@interfaceObject"])
{
  query: Query
}

directive @link(url: String, as: String, for: link__Purpose, import: [link__Import]) repeatable on SCHEMA

directive @key(fields: federation__FieldSet!, resolvable: Boolean = true) repeatable on OBJECT | INTERFACE

directive @requires(fields: federation__FieldSet!) on FIELD_DEFINITION

directive @provides(fields: federation__FieldSet!) on FIELD_DEFINITION

directive @external(reason: String) on OBJECT | FIELD_DEFINITION

directive @tag(name: String!) repeatable on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION | SCHEMA

directive @extends on OBJECT | INTERFACE

directive @shareable repeatable on OBJECT | FIELD_DEFINITION

directive @inaccessible on FIELD_DEFINITION | OBJECT | INTERFACE | UNION | ARGUMENT_DEFINITION | SCALAR | ENUM | ENUM_VALUE | INPUT_OBJECT | INPUT_FIELD_DEFINITION

directive @override(from: String!, label: String) on FIELD_DEFINITION

directive @composeDirective(name: String) repeatable on SCHEMA

directive @interfaceObject on OBJECT

directive @federation__authenticated on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM

directive @federation__requiresScopes(scopes: [[federation__Scope!]!]!) on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM

directive @federation__policy(policies: [[federation__Policy!]!]!) on FIELD_DEFINITION | OBJECT | INTERFACE | SCALAR | ENUM

directive @federation__sourceAPI repeatable on SCHEMA

directive @federation__sourceType repeatable on OBJECT | INTERFACE

directive @federation__sourceField repeatable on FIELD_DEFINITION

directive @federation__context(name: String!) repeatable on INTERFACE | OBJECT | UNION

directive @federation__fromContext(field: federation__ContextFieldValue) on ARGUMENT_DEFINITION

directive @federation__cost(weight: Int!) on ARGUMENT_DEFINITION | ENUM | FIELD_DEFINITION | INPUT_FIELD_DEFINITION | OBJECT | SCALAR

directive @federation__listSize(assumedSize: Int, slicingArguments: [String!], sizedFields: [String!], requireOneSlicingArgument: Boolean = true) on FIELD_DEFINITION

enum link__Purpose {
  """
  `SECURITY` features provide metadata necessary to securely resolve fields.
  """
  SECURITY

  """
  `EXECUTION` features provide metadata necessary for operation execution.
  """
  EXECUTION
}

scalar link__Import

scalar federation__FieldSet

scalar federation__Scope

scalar federation__Policy

scalar federation__ContextFieldValue

interface I {
  id: ID!
  x: Int
  y: Int
}

type C
  @key(fields: "id")
{
  id: ID! @shareable
  i: I @external
  c: String @requires(fields: "i { x y }")
}

scalar _Any

type _Service {
  sdl: String
}

union _Entity = C

type Query {
  _entities(representations: [_Any!]!): [_Entity]!
  _service: _Service!
}
```

Reply:
this ws an issue in the way I wrote mocks, the request was actually going to the starstuff subgraph